applicationinsights-react-native/src/ReactNativePlugin.ts

```diff
@@ -48,34 +55,42 @@ export class ReactNativePlugin extends BaseTelemetryPlugin {
                 core?: IAppInsightsCore,
                 extensions?: IPlugin[]
             ) => {
+                let identifier = this.identifier;
                 if (!_self.isInitialized()) {
                     _base.initialize(config, core, extensions);
 
-                    const inConfig = config || {};
-                    const defaultConfig = _getDefaultConfig();
-                    objForEachKey(defaultConfig, (option, value) => {
-                        _config[option] = ConfigurationManager.getConfig(
-                            inConfig as any,
-                            option,
-                            _self.identifier,
-                            !isUndefined(_config[option]) ? _config[option] : value
-                        );
-                    });
-        
-                    if (!_config.disableDeviceCollection) {
-                        _self._collectDeviceInfo();
-                    }
-        
-                    if (core && core.getPlugin) {
-                        _analyticsPlugin = core.getPlugin<any>(AnalyticsPluginIdentifier)?.plugin as IAppInsights;
-                    }
-        
-                    if (!_config.disableExceptionCollection) {
-                        _self._setExceptionHandler();
-                    }
+                    _self._addHook(onConfigChange(config, (details) => {
+                        let ctx = _self._getTelCtx();
+                        _config = ctx.getExtCfg<IReactNativePluginConfig>(identifier, defaultReactNativePluginConfig);
+
+                        if (!_config.disableDeviceCollection) {
+                            if (!deviceInfoCollected){
+                                _self._collectDeviceInfo();
+                                deviceInfoCollected = true;
+                            }
+                        } else {
+                            _deviceInfoModule = null;
```

Comment:
We probably need to have a "_userDeviceInfoModule" in the set and then reference that.

Reply:
I create a boolean value _deviceInfoNeedsUpdate so that as soon as the setting func get called, the onConfigChange will be called as well. And that would help reduce unnecessary called for coolectDeviceInfo. Hopefully this could work.